grin/src/Grin/ExtendedSyntax/Research.hs

```diff
@@ -0,0 +1,204 @@
+-- Based on the https://brianmckenna.org/blog/type_annotation_cofree
+--
+-- This module is an example how one can generate constraints based
+-- on the GRIN AST. Solve the constraints and annotate the nodes
+-- with some information.
+--
+-- Thus this modules serves as a fundation for quick prototyping and
+-- a tool which enables to create simple test oracle for DataFlow IR
+-- based abstract interpretation.
+
+{-# LANGUAGE TemplateHaskell #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE QuasiQuotes #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE DeriveFunctor #-}
+module Grin.ExtendedSyntax.Research where
+
+{- QUESTION: is this module needed?
```

Comment:
Nope. The probably the souffle approach would be good for research as well.

Reply:
Removed this and `Model.hs` as well.